### PR TITLE
Issue #162: `find` based on Candidate's availability

### LINKED ISSUE #162: Update `find` command for `Candidate`'s Availability

### PR DESCRIPTION
**Pull request by domlimm**
This PR consists of the following changes:
- Added additional field `availability` for user to search for when using the `find` command
- Added supporting test cases
- Fix minor bugs

Thanks @charmainehly for creating the issue!

Closes #162.

### CODE REVIEW

**Comment by domlimm** [comment]
@charmainehly @tiewweijian please hang on, fixing up some codecov stuff.

File: src/main/java/seedu/address/logic/commands/FindCommand.java [hunk 1]
```diff
@@ -19,8 +19,8 @@
 
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all candidates whose description contain any of "
             + "the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
-            + " The search will be conducted only on a specific field in candidates' description by specifying the"
-            + PREFIX_FIELD + " FIELD argument.\n"
+            + "The search will be conducted only on a specific field in candidates' description by specifying the "
```
Comment (charmainehly) [inline src/main/java/seedu/address/logic/commands/FindCommand.java:22]: Thanks for helping to fix the whitespacing 😄

**Review by charmainehly — APPROVED** [review]
LGTM!

File: src/main/java/seedu/address/logic/commands/FindCommand.java (hunk 1 above)
Comment (domlimm) [reply src/main/java/seedu/address/logic/commands/FindCommand.java:22]: No problem at all! Char wait! More stuff to add

**Review by charmainehly — APPROVED** [review]

**Review by tiewweijian — APPROVED** [review]
LGTM ! Thanks Dom !!

**Pull request merged**